src/ast.ts

```diff
@@ -830,6 +830,14 @@ export abstract class TypeNode extends Node {
     super(kind, range);
   }
 
+  isNullType(): bool {
+    if (this.kind == NodeKind.NAMEDTYPE) {
+      let namedTypeNode = <NamedTypeNode>changetype<TypeNode>(this); // TS
+      return namedTypeNode.name.identifier.text == "null";
+    }
+    return false;
+  }
```

Comment:
I think it should be in `Node` instead of `TypeNode`. Because you're anyway internally downcast to `NamedTypeNode`. Also, `TypeNode` already has `isNullable` which duplicate `isNullType`

Reply:
static is a good idea. But add it into node is a little strange. Because in most case at least now we only want to check whether a `TypedNode` is `NullType` instead of `Node`. Add it into `Node` scope will make other people very hard to find and reuse it. But of course the argument can be a `Node`.

Reply:
> static is a good idea. But add it into node is a little strange. Because in most case at least now we only want to check whether a TypedNode is NullType instead of Node

In this case, better move it to sole util function due to it uses only in parser.ts